services/user_service/database_connector.py

```diff
@@ -0,0 +1,110 @@
+import firebase_admin
+from firebase_admin.exceptions import FirebaseError
+from firebase_admin import credentials
+from firebase_admin import db
+
+
+class DatabaseAdmin:
+
+    def __init__(self,databaseURL, credentials_path):
+        cred = credentials.Certificate(credentials_path)
+        firebase_admin.initialize_app(cred,
+        {
+            'databaseURL' : databaseURL,
+        })
+        self.__users_ref = db.reference("users")
+
+    def get_all_users(self):
+        try:
+            if self.__users_ref.get() is None:
+                return list()
+            return self.__users_ref.get()
+        except FirebaseError as e:
+            print("###Connection lost. EROR received : " + str(e))
+            return list()
+    
+
+    def get_user_by_id(self, id):
+        try:
+            if self.__users_ref.child(id).get() is None:
+                return list()
+            return self.__users_ref.child(id).get()
+        except ValueError as e:
+            print("###EROR received : " + str(e))
+        except FirebaseError as e:
+            print("###Connection lost. EROR received : " + str(e))
+        return list()
+
+
+    def get_user_by_email(self, email):
+        try:   
+            for user_id in self.get_user_ids():
+                if email == self.__users_ref.child(user_id).child("email").get():          
+                    return self.__users_ref.child(user_id).get()
+        except ValueError as e:
+            print("###EROR received : " + str(e))
```

Comment:
What about [logging library in python](https://docs.python.org/3/library/logging.html)?

Reply:
I'm not sure we need it right now. We do not expect to procces a megabytes of logs at this point. We will add logger separately if needed later.

Reply:
When we deploy this and your server failed, how will you understand what happened there?

Reply:
I will read logs in console)

Reply:
you don't have any access to console. especially, if your service is down

Reply:
it prints standard output as well as error messages

Reply:
I don't think it will work in remote, but let it be.